### DISCUSSION TITLE
Vorta not showing up in macOS notifications?

### DISCUSSION THREAD

**Discussion by MacH59-cos** [borgbase/vorta#1272] [General]
Recently with i believe version 0.82 and 0.83 of Vorta on BigSur and Monterey Vorta is not immediately showing up in the macOS notifications panel. Have not really tested it with older version of macOS. I believe that it later will appear by itself but not sure what triggers this.

Is really quite annoying since I can not finish the installation by setting up the warning system through notifications in the best way possible. (One of the best parts of Vorta if you ask me.) Used to be not a problem ie Vorta showed up directly.

Any idea how this can be fixed? I now have to remember and go back tot that Mac to later 'correct' those setting. Again: very annoying.

**Comment by m3nu** [comment]
Couldn't reproduce this with the released 0.8.3 app bundle. Maybe you have disabled the setting under *Misc* or in macOS itself?

![Screen Shot 2022-04-26 at 14 50 07](https://user-images.githubusercontent.com/3916435/165284462-77975e2c-f06d-4990-811d-ceaf4195a512.jpg)

**Comment by MacH59-cos** [comment]
No, is chosen in Misc (both failure and normal activity) and in macOS itself I cannot select anything since it is not showing up in notifications panel. Note: this happens when you do a new install on a mac where Vorta was not installed before. And again I believe that later (maybe after initial backup is done??) suddenly Vorta does appear in the notifications panel - not a 100% sure about that. But this is not very handy because as a said before I can not do all setting at once and change the standard 'banners' type into 'alerts' type that will stick in the corner until the user closes it. Banners just appear very quickly and are easily overlooked.

That way there is extra certainty about whether Borg is running correctly, which is very nice to have.

**Reply by real-yfprojects** [reply]
This is probably because vorta only connects to the native MacOS notification system if a notification should be displayed. Therefore
vorta will not show up in the Notifications previous to that. Also vorta still uses a deprecated API for showing notifications on MacOS. It therefore doesn't support the new MacOS way of configuring notifications. I am actually already working on implementing the new API.

**Comment by MacH59-cos** [comment]
Thank you for all your answers! 

One last remark from my side: It should probably be possible to change the standard 'banner' warning type into the 'alert' type for Vorta when it gets activated for the first time. That way the safest warning type is choosen as standard and if somebody does not like that he can than change it to banner again... Banners, as i said, are easily overlooked since they are only visible for a short moment.

Any chance this could be implemented?

**Comment by MacH59-cos** [comment]
Hi, I just recently installed/used an older Vorta version 0.6.23 on a Macintosh with High Sierra on it (none of the newer versions work on High Sierra) The Mac did not have any prior installation of Vorta on it. To my surprise and delight Vorta did immediately show up in the 'Notifications' Panel of MacOS. As I  mentioned above this is NOT the case anymore with any later versions of Vorta. It will take a certain amount of time before Vorta appears in the 'Notifications' Panel of MacOS. Although this is not a critical problem it is very frustrating since I can not directly at the moment of installation finish this last setting (and change the standard 'banners' type into 'alerts' type) which is very useful for controlling if the backups do run regularly. You then do not have to check the archive list to do that. And the reminder is done automatically instead of the user looking at the archive list.

As version 0.6.23 clearly shows: this is very wel possible and I would say the correct behavior if an Application can make use of 'Notifications'. It is probably important to do any testing on a Mac without any prior Vorta installation. I assume if there was any prior installation Vorta might be already there in the 'Notifications' Panel of MacOS because of that.

Is it possible that someone of the creators of Vorta can look into this please? It would be very much appreciated.

Thank you.

**Reply by m3nu** [reply]
High Sierra has been EOL for a long time. Hope you don't connect this machine to the internet.

I doubt anyone here wants to spend their time toying with outdated macOS versions. You're still free to grab the source code and build it yourself. That's the beauty of open source. 😏

**Reply by real-yfprojects** [reply]
Related to MacOs Notifications:

The corresponding API was renewed recently and vorta still uses the deprecated one. Though I have already written some patches for migrating to the new API.

**Comment by MacH59-cos** [comment]
m3nu, you did not read my question correctly...

I am not asking  to 'repair' something in an old version of Vorta or macOS. I wanted to let you know that something that was working in that version is not working ever since. I doubt that something did change fundamentally with 'Notifications' Panel of MacOS since High Sierra and my notifications panel is full with all kinds of 3rd party apps. As far as I can tell those appear immediately after a program is installed. It looks to me that that is what i would call 'normal' behavior.

I or clients of mine use even older systems that are of course connected to the internet. If you use proper safety measures (like a good Anti-Virus program) that is NO problem. You should not believe all the BS that Apple wants us to believe. After all macOS is UNIX and there are not many known threats for macOS. Apple is just using this to scare people into upgrading there Mac's so that certain older Mac's become unusable and those users wil buy a new Mac again...

Trust me: I have 25 years of experience with supporting Mac's and Mac users.

**Reply by m3nu** [reply]
My Vorta notifications work well on macOS 12.

**Reply by MacH59-cos** [reply]
(I am afraid I haven't been clear enough so one 'last' try please:)

That is also not the 'problem'. Notifications, once they are active, work very well. As a matter of fact it is a brilliant way of letting users know their backups are running. But the best way with this notification system is to change the Vorta standard 'banners' type into 'alerts' type that will stay in the corner until it is manually dismissed by a user.

You can only change that setting if Vorta is listed in the notification center.That is were the 'problem' lies: with all current and new Vorta installers Vorta will NOT show up in the notifications center immediately after installation. It will appear at a later time but i havent been able to  find out wat is the cause (maybe after initial backup is finished?).

Anyway as the older Vorta version 0.6.23 shows this was not always the case. With that version Vorta is immediately available in the notification center after installation. This is the same behavior that other 3rd party apps show. If an app can work with notifications center and you install it it will immediately be available in notifications center.

So do you have any explanation why this is not working anymore in this way with Vorta?? Is there a technical reason for this maybe?

Do you think it can be fixed?...

**Reply by MacH59-cos** [reply]
Forgot one thing: the best solution as far as I am concerned would be if Vorta by default would use the 'alerts' type instead of the standard 'banners' type. But I don't know if that can be done by the Vorta installer...

That way it does not matter if it is listed immediately after installation. If a user does not like those alerts he can then change it himself into the standard 'banners' type.

Is this possible?

**Reply by m3nu** [reply]
Definitely possible to make this change. Feel free to submit a pull request when ready and we'll look into it. Maybe together with updating the notification API.

**Reply by m3nu** [reply]
Notification code is pretty simple and found here: https://github.com/borgbase/vorta/blob/master/src/vorta/notifications.py#L45

**Reply by real-yfprojects** [reply]
> You can only change that setting if Vorta is listed in the notification center.That is were the 'problem' lies: with all current and new Vorta installers Vorta will NOT show up in the notifications center immediately after installation. It will appear at a later time but i havent been able to find out wat is the cause (maybe after initial backup is finished?).

Probably after the first notification was sent by vorta.

> Anyway as the older Vorta version 0.6.23 shows this was not always the case. With that version Vorta is immediately available in the notification center after installation. 

This is weird since there was no change to the notifications code since version `v0.6.5`. Though packaging was adjusted with version `v.6.24`. Can you try whether it works with this version?

> So do you have any explanation why this is not working anymore in this way with Vorta?? Is there a technical reason for this maybe?
> Do you think it can be fixed?...

I always guessed it is because of a new version of the `Objective-C` framework used to interact with MacOS. Therefore it would be fixed by updating the code to use the new version.
